File: BillingSystem.Sales/Domain/ValueObjects/Quantity.cs
namespace BillingSystem.Sales.Domain.ValueObjects;

public class Quantity
{
    public decimal Value { get; }

    public Quantity(decimal value)
    {
        if (value < 0)
            throw new ArgumentException("Quantity cannot be negative.");
        Value = value;
    }

    public static Quantity operator +(Quantity q1, Quantity q2)
    {
        return new Quantity(q1.Value + q2.Value);
    }

    public static Quantity operator -(Quantity q1, Quantity q2)
    {
        return new Quantity(q1.Value - q2.Value);
    }

    public override string ToString() => Value.ToString();
}

File: BillingSystem.Sales/Client/src/app/models/Quantity.ts
export class Quantity {
    public value: number;

    constructor(value: number) {
        if (value < 0) throw new Error("Quantity cannot be negative.");
        this.value = value;
    }

    public static add(q1: Quantity, q2: Quantity): Quantity {
        return new Quantity(q1.value + q2.value);
    }

    public static subtract(q1: Quantity, q2: Quantity): Quantity {
        return new Quantity(q1.value - q2.value);
    }
}
